### PR TITLE
add string#blank?

### PR DESCRIPTION
**Pull request by kostya**

### CODE REVIEW

**Comment by jhass** [comment]
I have to -1 this. I think how "blankness" is defined is highly domain/application specific, that's why it remains in ActiveSupport in Ruby land, an extension library supporting a *Web*Framework.

**Comment by kirbyfan64** [comment]
I would prefer a different name, something like Python `isspace`. What about `ws?`?

**Comment by jhass** [comment]
`ws` is cryptic, `whitespace?` could be acceptable. Though I'm not sure it's clearly defined either and considering the empty string whitespace doesn't feel particular right either.

**Comment by kirbyfan64** [comment]
Does `Char` have a `whitespace?` function? If not, then it could be added there. If someone wanted to test if a string was whitespace, they'd need to do something like `str.all? &.whitespace?`, so the semantics are more clearly defined.

I personally end up usually checking if a single character is whitespace, anyway.

**Comment by kostya** [comment]
`each_char.all? &.blank?` twice as slower than each_char {}, why?

**Comment by asterite** [comment]
Because `each_char` returns an Iterator, which is a class, so that allocates memory

**Comment by asterite** [comment]
and also I think LLVM will find it harder to optimize that code

**Comment by asterite** [comment]
By the way, @waj and I think `blank?` is good and we might accept this.

We could enhance the current `Char#whitespace?` to take the other codepoints into account.

**Comment by kostya** [comment]
can it make, for example json parser, slower?

**Comment by asterite** [comment]
@kostya I just checked it and just a bit slower. But JSON shouldn't use `char.whitespace?`: http://stackoverflow.com/a/16042321/4990850

**Comment by PragTob** [comment]
:+1: I like the inclusion of `blank?` :)

File: src/string.cr [hunk 1]
```diff
@@ -2446,6 +2450,7 @@ class String
     include Iterator(Char)
 
     def initialize(@reader, @end = false)
+      @end = !@reader.has_next?
```
Comment (lbguilherme) [inline src/string.cr]: Is this related to `blank?`'s implementation?

Comment (kostya) [inline src/string.cr]: not really, but chariterator raised on empty strings, and crashed my specs

**Comment by asterite** [comment]
We've been discussing this with @waj and we think this should go to a `Unicode` module that should answer all these questions (char is blank, string is blank, etc.)
